src/sql/connection.py

```diff
@@ -359,27 +382,53 @@ def assign_name(cls, engine):
         return name
 
     @classmethod
-    def connection_list(cls):
-        """Returns the list of connections, appending '*' to the current one"""
-        result = []
+    def _get_connections(cls):
+        """
+        Return a list of dictionaries
+        """
+        connections = []
+
         for key in sorted(cls.connections):
             conn = cls.connections[key]
 
-            if cls.is_custom_connection(conn):
-                engine_url = conn.url
-            else:
-                engine_url = conn.metadata.bind.url if IS_SQLALCHEMY_ONE else conn.url
+            current = conn == cls.current
 
-            prefix = "* " if conn == cls.current else "  "
+            connections.append(
+                {
+                    "current": current,
+                    "key": key,
+                    "url": conn.url,
+                    "alias": conn.alias,
+                    "connection": conn,
+                }
+            )
 
-            if conn.alias:
-                repr_ = f"{prefix} ({conn.alias}) {engine_url!r}"
-            else:
-                repr_ = f"{prefix} {engine_url!r}"
+        return connections
 
-            result.append(repr_)
+    @classmethod
+    def display_current_connection(cls):
+        for conn in cls._get_connections():
+            if conn["current"]:
+                alias = conn.get("alias")
+                if alias:
+                    display.message(f"Running query in {alias!r}")
+                else:
+                    display.message(f"Running query in {conn['url']!r}")
```

Comment:
custom engines might won't have the `url` attribute (e.g. `pg.connect()`).
When we run a custom connection without an alias it will print:

![image](https://github.com/ploomber/jupysql/assets/103026689/30d383b9-d307-454f-9305-ffc7ea352f3d)

The connections are:
![image](https://github.com/ploomber/jupysql/assets/103026689/c3e20e73-e891-482c-8177-13c85a7ece9d)

Reply:
I forgot about custom connections. What did we print before when listing connections?

Reply:
![image](https://github.com/ploomber/jupysql/assets/103026689/8ef9dc78-d252-48a6-bfcd-b9ceb054243f)

Reply:
great catch! I realized the problem is that I wasn't escaping the HTML characters.